### PR TITLE
Snappi sample

### PR DESCRIPTION
**Pull request by ANISH-GOTTAPU**
### Description of PR
Migration of ixa/test_tgen from `ixnetwork open traffic generator` to `snappi`

Summary:
Test case(new/improvement)

### Type of change
- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x ] Test case(new/improvement)

#### What is the motivation for this PR?
The motivation for this PR is to migrate the tests that are written with `ixnetwork open traffic generator` to `snappi`

#### How did you do it?
Replaced the code where abstract is used to snappi

#### How did you verify/test it?
verified the tests in sonic-mgmt using pytest with DUT and IXIA connections 
snappi version used for the development of tests `snappi[ixnetwork]==0.4.14`

### CODE REVIEW

**Comment by ANISH-GOTTAPU** [comment]
This PR is dependant on https://github.com/Azure/sonic-buildimage/pull/7829

File: tests/common/fixtures/snappi_fixtures.py [hunk 1]
```diff
@@ -0,0 +1,413 @@
+"""
+This module contains the snappi fixture
+"""
+import pytest
+import snappi
+from ipaddress import ip_address, IPv4Address
+from tests.common.fixtures.conn_graph_facts import conn_graph_facts,\
+    fanout_graph_facts
+from tests.common.ixia.common_helpers import get_vlan_subnet, get_addrs_in_subnet,\
```
Comment (baiwei0427) [inline tests/common/fixtures/snappi_fixtures.py:9]: Can you create a new folder test.common.snappi to hold all the helper functions used to snappi?

File: tests/common/fixtures/snappi_fixtures.py [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+    get_peer_ixia_chassis
+from tests.common.ixia.ixia_helpers import IxiaFanoutManager, get_tgen_location
+from tests.common.ixia.port import IxiaPortConfig, IxiaPortType
+from tests.common.helpers.assertions import pytest_assert
+
+
+@pytest.fixture(scope="module")
+def snappi_api_serv_ip(tbinfo):
+    """
+    In a tgen testbed, there is no PTF docker.
```
Comment (baiwei0427) [inline tests/common/fixtures/snappi_fixtures.py:19]: tgen -> snappi

File: tests/common/fixtures/snappi_fixtures.py [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+    """
+    try:
+        return True if type(ip_address(ip)) is IPv4Address else False
+    except ValueError:
+        return False
+
+
+def __l3_intf_config(config, port_config_list, duthost, ixia_ports):
+    """
+    Generate Tgen configuration of layer 3 interfaces
```
Comment (baiwei0427) [inline tests/common/fixtures/snappi_fixtures.py:94]: Tgen -> Snappi

File: tests/common/fixtures/snappi_fixtures.py [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+        ethernet.name = 'Ethernet Port {}'.format(port_id)
+        ethernet.mac = mac
+
+        ip_stack = ethernet.ipv4
+        ip_stack.name = 'Ipv4 Port {}'.format(port_id)
+        ip_stack.address = ip
+        ip_stack.prefix = prefix
+        ip_stack.gateway = gw_addr
+
+        port_config = IxiaPortConfig(id=port_id,
```
Comment (baiwei0427) [inline tests/common/fixtures/snappi_fixtures.py:147]: IxiaPortConfig -> SnappiPortConfig

File: tests/snappi/test_snappi.py [hunk 5]
```diff
@@ -0,0 +1,188 @@
+import logging
+import time
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert, pytest_require
+from tests.common.fixtures.conn_graph_facts import conn_graph_facts,\
+    fanout_graph_facts
+from tests.common.fixtures.snappi_fixtures import snappi_api_serv_ip, snappi_api_serv_port,\
+    snappi_api, tgen_testbed_config
+from tests.common.ixia.port import select_ports
```
Comment (baiwei0427) [inline tests/snappi/test_snappi.py:10]: ixia -> snappi

File: tests/snappi/test_snappi.py [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+            flow.rate.percentage = rate_percent
+            flow.duration.fixed_seconds.seconds = duration_sec
+
+            flow.metrics.enable = True
+            flow.metrics.loss = True
+
+    return testbed_config
+
+
+def test_snappi(snappi_api,
```
Comment (baiwei0427) [inline tests/snappi/test_snappi.py:81]: Align all the arguments

**Review by baiwei0427 — CHANGES_REQUESTED** [review]

**Comment by ANISH-GOTTAPU** [comment]
updated lab.yml to take `snappi_api_server` instead of `ixia_api_server` 

ref: 
https://github.com/ANISH-GOTTAPU/sonic-testbed-manager/blob/snappi/ixia-calbases-lab/ansible/group_vars/lab/lab.yml#L52

File: tests/snappi/test_snappi.py (hunk 6 above)
Comment (ANISH-GOTTAPU) [reply tests/snappi/test_snappi.py:80]: updated

File: tests/snappi/test_snappi.py (hunk 5 above)
Comment (ANISH-GOTTAPU) [reply tests/snappi/test_snappi.py]: updated

File: tests/common/fixtures/snappi_fixtures.py (hunk 4 above)
Comment (ANISH-GOTTAPU) [reply tests/common/fixtures/snappi_fixtures.py]: Created new file in snappi folder and updated the class name

File: tests/common/fixtures/snappi_fixtures.py (hunk 3 above)
Comment (ANISH-GOTTAPU) [reply tests/common/fixtures/snappi_fixtures.py]: updated

File: tests/common/fixtures/snappi_fixtures.py (hunk 2 above)
Comment (ANISH-GOTTAPU) [reply tests/common/fixtures/snappi_fixtures.py]: updated

File: tests/common/fixtures/snappi_fixtures.py (hunk 1 above)
Comment (ANISH-GOTTAPU) [reply tests/common/fixtures/snappi_fixtures.py]: created new folder and moved helper files as suggested

File: tests/snappi/test_snappi.py [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+from tests.common.helpers.assertions import pytest_assert, pytest_require
+from tests.common.fixtures.conn_graph_facts import conn_graph_facts,\
+    fanout_graph_facts
+from tests.common.snappi.snappi_fixtures import snappi_api_serv_ip, snappi_api_serv_port,\
+    snappi_api, snappi_testbed_config
+from tests.common.snappi.port import select_ports
+from tests.common.snappi.qos_fixtures import prio_dscp_map
+
+
+@pytest.mark.topology("tgen")
```
Comment (baiwei0427) [inline tests/snappi/test_snappi.py:13]: change "tgen" to "snappi"

**Review by baiwei0427 — CHANGES_REQUESTED** [review]

File: tests/snappi/test_snappi.py (hunk 7 above)
Comment (ANISH-GOTTAPU) [reply tests/snappi/test_snappi.py]: changed

**Comment by baiwei0427** [comment]
@ANISH-GOTTAPU Can you fix LGTM alerts?

**Review by baiwei0427 — APPROVED** [review]

**Comment by baiwei0427** [comment]
@neethajohn  Can you merge this PR?

**Review by neethajohn — APPROVED** [review]

**Comment by bingwang-ms** [comment]
Sorry @neethajohn @ANISH-GOTTAPU , I have to revert this PR since it causes regression. Please check #3766 for more info. Thanks

**Pull request merged**